system-parachains/collectives/collectives-polkadot/src/secretary/mod.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The Polkadot Secretary Collective.
+
+mod origins;
+use core::marker::PhantomData;
+
+use crate::{fellowship::FellowshipAdminBodyId, *};
+use frame_support::{
+	parameter_types,
+	traits::{tokens::GetSalary, EitherOf, MapSuccess, PalletInfoAccess, PollStatus, Polling},
+};
+use frame_system::{pallet_prelude::BlockNumberFor, EnsureRootWithSuccess};
+pub use origins::{pallet_origins as pallet_secretary_origins, Secretary};
+use pallet_ranked_collective::{MemberIndex, TallyOf, Votes};
+use pallet_xcm::{EnsureXcm, IsVoiceOfBody};
+use polkadot_runtime_constants::time::HOURS;
+use sp_core::{ConstU128, ConstU32};
+use sp_runtime::{
+	traits::{ConstU16, ConvertToValue, Identity, Replace},
+	DispatchError,
+};
+
+use xcm::prelude::*;
+use xcm_builder::{AliasesIntoAccountId32, PayOverXcm};
+
+use self::xcm_config::AssetHubUsdt;
+
+/// The Secretary members' ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	pub const SECRETARY_CANDIDATE: Rank = 0;
+	pub const SECRETARY: Rank = 1;
+}
+
+/// Origins of:
+/// - Root;
+/// - FellowshipAdmin (i.e. token holder referendum);
+/// - Plurality vote from Fellows can promote, demote, remove and approve rank retention
+/// of members of the Secretary Collective (rank `2`).
+type ApproveOrigin = EitherOf<
+	EnsureRootWithSuccess<AccountId, ConstU16<65535>>,
+	EitherOf<
+		MapSuccess<
+			EnsureXcm<IsVoiceOfBody<GovernanceLocation, FellowshipAdminBodyId>>,
+			Replace<ConstU16<65535>>,
+		>,
+		MapSuccess<Fellows, Replace<ConstU16<65535>>>,
+	>,
+>;
+
+impl pallet_secretary_origins::Config for Runtime {}
+
+pub struct SecretaryPolling<T: pallet_ranked_collective::Config<I>, I: 'static>(
+	PhantomData<(T, I)>,
+);
+
+impl<T: pallet_ranked_collective::Config<I>, I: 'static> Polling<TallyOf<T, I>>
```

Comment:
it would be nice to get this no-op impl for unit or some type in frame, where the trait is defined. we can leave a TODO here with the link to polkadot-sdk PR if you open it.

Reply:
why would we need this in `pallet_ranked_collective::tests`?